protocol_rfcs/column-mapping-usage-tracking.md

```diff
@@ -0,0 +1,43 @@
+
+--------
+
+> New subsection at the end of the `Column Mapping` section
+
+## Usage Tracking
+
+Column Mapping Usage Tracking is an extension of the column mapping feature that allows Delta to track whether a column has been dropped or renamed.
+This is tracked by the table property `delta.columnMapping.hasDroppedOrRenamed`. This table property is set to `false` when the table is created, and flipped to `true` when the first column is either dropped or renamed.
+The writer table feature `columnMappingUsageTracking` is added to the `writerFeatures` in the `protocol` to ensure that all writers correctly track when columns are dropped or renamed.
+
+--------
+
+> Modification to the `Writer Requirements for Column Mapping` subsection
+
+- Assign a globally unique identifier as the physical name for each new column that is added to the schema. This is especially important for supporting cheap column deletions in `name` mode. In addition, column identifiers need to be assigned to each column. The maximum id that is assigned to a column is tracked as the table property `delta.columnMapping.maxColumnId`. This is an internal table property that cannot be configured by users. This value must increase monotonically as new columns are introduced and committed to the table alongside the introduction of the new columns to the schema.
+
+**is replaced by**
+
+- Assign a unique physical name to each column.
+    - When enabling column mapping on existing table, the physical name of the column must be set to the (logical) name of the column.
+    - If the feature `columnMappingUsageTracking` is supported, then when adding a new column to a table and `delta.columnMapping.hasDroppedOrRenamed` column property is `false` the (logical) name of the column should be used as the physical name.
+    - Otherwise the physical column name must contain a universally unique identifier (UUID) to guarantee uniqueness.
+- Assign a column id to each column. The maximum id that is assigned to a column is tracked as the table property `delta.columnMapping.maxColumnId`. This is an internal table property that cannot be configured by users. This value must increase monotonically as new columns are introduced and committed to the table alongside the introduction of the new columns to the schema.
+
+--------
+
+> New subsection at the end of the `Writer Requirements for Column Mapping` subsection
+
+### Writer Requirements for Usage Tracking
+
+In order to support column mapping usage tracking, writers must:
+- Write `protocol` and `metaData` actions when Column Mapping Usage Tracking is turned on for the first time:
+    - Write a `protocol` action with writer version 7 and the feature `columnMappingUsageTracking` in the `writerFeatures`.
+    - Write a `metaData` action with the table property `delta.columnMapping.hasDroppedOrRenamed` set to `false` when creating a new table, or set to `true` when enabling usage tracking on an existing table.
```

Comment:
Enabling Usage Tracking for existing tables where column mapping is disabled also requires setting `delta.columnMapping.hasDroppedOrRenamed` to true?

Reply:
@tomvanbussel